### ISSUE TITLE
Obsolete diagnostics not reported for delegate creation

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#47308]
```
using System;
public class C {
    [Obsolete]
    public void M() {
    }
    
    void M2()
    {
        Action a = M; // Diagnostic reported
        a = new Action(M); // No diagnostic reported
    }
}
```

**Comment by Youssef1313** [comment]
The two warnings existed in the native compiler.
Not sure if this bug existed from the beginning of Roslyn or this is a regression at some Roslyn version.
